### PR TITLE
[MASSEMBLY-964] Fix initFile for FileLocation, UrlLocation

### PR DESCRIPTION
**Pull request by slawekjaranowski**
method FileLocation.initFile was private and can not be override in child class

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MASSEMBLY) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MASSEMBLY-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MASSEMBLY-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ File unsafeGetFile()
      * initialize file.
      * @throws IOException in case of error
      */
-    private void initFile()
+    protected void initFile()
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/assembly/io/FileLocation.java:110]: wouldn't @Override annotation help?

File: src/main/java/org/apache/maven/plugins/assembly/io/URLLocation.java [hunk 2]
```diff
@@ -60,7 +60,7 @@
     }
 
     /** {@inheritDoc} */
-    void initFile()
+    protected void initFile()
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/assembly/io/URLLocation.java:63]: wouldn't `@Override` annotation be a better choice than `{@inheritDoc}` javadoc

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/assembly/io/URLLocation.java:63]: `@Override` added

File: pom.xml [hunk 3]
```diff
@@ -365,6 +365,7 @@ under the License.
                 <https.protocols>${https.protocols}</https.protocols>
               </properties>
               <showVersion>true</showVersion>
+<!--              <parallelThreads>1C</parallelThreads>-->
```
Comment (hboutemy) [inline pom.xml:368]: I suppose this comment should be removed

Comment (slawekjaranowski) [reply pom.xml]: my local testing ... removed

**Review by cstamas — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1172

**Pull request merged**